File: BackJoon/Silver/BS5_set.js
var fs = require('fs');
// let input = fs.readFileSync('/dev/stdin').toString().trim().split('\n');
var input = fs.readFileSync('예제.txt').toString().trim().split('\r\n');
let answer = [];
let arr = [];
input = input.slice(1);
input.forEach((v, i) => {
    let [string, num] = v.split(" ");
    switch (string) {
        case "add":
            arr[num] = 1;
            break;
        case "remove":
            arr[num] = 0;
            break;
        case "check":
            arr[num] == 1 ? answer.push(1) : answer.push(0);
            break;
        case "toggle":
            arr[num] == 1 ? arr[num] = 0 : arr[num] = 1;
            break;
        case "all":
            for (let j = 1; j <= 20; j++)
                arr[j] = 1;
            break;
        case "empty":
            arr.length = 0;
            break;

    }
})

console.log(answer.join("\n"));

// js 메모리 설정이 없어서 메모리초과만 뜸

File: BackJoon/Silver/BS5_set.java
// 2022-06-09
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

        StringBuilder sb = new StringBuilder();
        StringTokenizer st = new StringTokenizer(br.readLine());
        int num = Integer.parseInt(st.nextToken());
        String answer = "";
        Boolean[] arr = new Boolean[21];
        for (int i = 0; i < num; i++) {
            st = new StringTokenizer(br.readLine());
            String word = st.nextToken();
            int number = 0;
            if (!word.equals("all") && !word.equals("empty")) {
                number = Integer.parseInt(st.nextToken());
            }
            switch (word) {
                case "add":
                    arr[number] = true;
                    break;
                case "remove":
                    arr[number] = false;
                    break;
                case "check":
                    sb.append(arr[number] != null && arr[number] ? "1\n" : "0\n");
                    break;
                case "toggle":
                    arr[number] = arr[number] == null ? true : !arr[number];
                    break;
                case "all":
                    for (int j = 1; j <= 20; j++)
                        arr[j] = true;
                    break;
                case "empty":
                    for (int j = 1; j <= 20; j++)
                        arr[j] = false;
                    break;
            }
        }
        bw.write(sb.toString());
        bw.flush();
        bw.close();
        br.close();
    }
}


// 자바는 자바스크립트에 비해서 상당히 세세하게 시간초과를 잡는거 같다
